### ISSUE TITLE
Raven core wallet passphrase minimum requirement on older version 2.2.2.0 and earlier

### ISSUE THREAD

**Issue by cupidchild** [RavenProject/Ravencoin#882]
Does anyone know if there was any minimum length requirement on older wallet passphrase that had to be created? Version 2.2.2.0
 and early version?

I remember creating some simple, one word passphrase on 0.15 version however after upgrading to a later version of Raven Wallet like 2.0.2 , the wallet passphrase does not work. So now I need to resort to trying to figure out the passphrase on it. I don't think it matter what version I am running since wallet.dat file is the same along with whatever passphrase it has in it.

I didn't find any information on wiki nor readme, that mentions what the minimum length has to be.

This is before the wallet supported 24 words password. So any information on password requirement would help.

thanks!

**Comment by RavencoinFoundationTron** [comment]
I am not aware of changes that were made that would prevent previous passwords from working.  However, I have heard from 2 people that have contacted me directly with a similar issue (or wrote down passwords wrong).  I will be meeting personally with one of them - likely later this week - to see what I can learn.

**Comment by cupidchild** [comment]
I have my wallet.dat file. and currently on Raven QT wallet  version 2.2.2.0-xxx   and all sync-ed up. 
I see the quantity of my ravencoin and  of course when I try to send coins to some address, that's when it askes for the passphrase of the wallet.      I am not going to upgrade my wallet to the latest, in fear that I will loose something.. lol 
I am in the process of building a PC with good GPU so that I can use  btcreovery program to crack the password.  

But please do let me know what you find.  
![image](https://user-images.githubusercontent.com/80020950/110364943-678d1d00-8012-11eb-990c-2aaade45863e.png)

**Comment by cupidchild** [comment]
Did you find out anything?  
I'm on Raven 2.2.2.0   should I try to downgrade or upgrade to latest ? *(And try whatever password that I remember)  I don't think this would make a difference.  
Let me know.
thanks

**Comment by TronBlack** [comment]
Try downgrading first and see if it works.

You will not be able to sync to the end, but then you'll know if the
password was working on an older version.


Tron

**Comment by cupidchild** [comment]
okay thx.. will try it on  v0.15.99.0   and then move onto v2.0.1

**Comment by cupidchild** [comment]
![image](https://user-images.githubusercontent.com/80020950/111085377-8b0e0700-84ed-11eb-987f-ca830abaa8e2.png)

v.0.15.99.0 failed to open. 
Nothing on debug.log
will try 2.0.1

**Comment by cupidchild** [comment]
v2.0.1 gave some interesting error: 

![image](https://user-images.githubusercontent.com/80020950/111085506-0a9bd600-84ee-11eb-9d51-46101c5a4f72.png)

**Comment by cupidchild** [comment]
v2.0.1 after the warning gave fatal error. 
Moving to v2.0.2

![image](https://user-images.githubusercontent.com/80020950/111086894-1dfe6f80-84f5-11eb-943e-8ad81c75c440.png)

**Comment by jeroz1** [comment]
This reads as if the wallet got corrupted along the way somewhere and recovery on it failed. You wouldn't be the first one. Do you have the original backup of the file somewhere?

**Comment by cupidchild** [comment]
when I open it up on v2.2.2.0  it opens up fine without any error. (*but not completely sure wallet.dat file is corrupt or not) 
I was trying to open it with previous version to see if I can unlock using the password that I know.

**Comment by jeroz1** [comment]
Does your passphrase work on 0.15.99? You can test the passphrase by unlocking your wallet in the debug console. You do not have to be all synced up to check this. 

To unlock:
help > debug window > console, and type:
`walletpassphrase YOURPASSPHRASEHERE 600`
This will unlock your wallet.dat file for 600 seconds. 

If your passphrase was correct, the wallet will respond with `"null"`. If that is the case, your wallet.dat file should work just fine when using the latest Ravencoin version. If the passphrase was incorrect, it will respond with an error. In that case either the passphrase is not correct. (I am unsure how the wallet would behave if the wallet.dat file is actually from a different bitcoin derived blockchain project, if that may be the case).

If you are able to unlock your wallet.dat file but it still doesn't work with newer versions for some reason, you can also dump your wallet to a text file to get access to your private keys. If you want to do so, you can do that in the same console after you unlocked your wallet, by typing:
`dumpwallet "wallet.txt"`
The console should respond where it saved the this text file. By default it should be in the same folder as where your wallet.dat file is stored.

**Comment by RavencoinFoundationTron** [comment]
I'm meeting with someone with a similar issue this afternoon.  I'll report
back.

**Comment by cupidchild** [comment]
yeah, this is same as if I were to send RVN to another address and it askes for the password. 
I did try cli via the debug window and of course the password didn't work. :(

**Comment by cupidchild** [comment]
I noticed something on the right side of the Raven core -wallet..  
This is after block time has all synced up to date. 

![image](https://user-images.githubusercontent.com/80020950/111203660-aafc0400-859b-11eb-90c6-56b1fa721dc3.png)

**Comment by cupidchild** [comment]
on another pc, I just ran latest Raven Core Wallet version 5.7.1 
and synced up and no issue nor error message...  
I see my coins qty. but of course just can't send them anywhere. lol

**Comment by TronBlack** [comment]
Update.  I did meet with a user that was unable to unlock their wallet.dat.  We installed an older version and tried again and it did not work so it is more likely the password was entered incorrectly originally.  At this point, there isn't enough evidence to suggest that code changes are preventing original passwords from unlocking wallet.dat files.  Please continue to report any issues that might suggest otherwise.  This will be left open for now.

**Comment by hans-schmidt** [comment]
Another month has passed since Tron's update with no information indicating that the fault is in the code.
Therefore, I am closing this issue. 
Please open a new issue if new relevant information becomes available.

**Comment by RyanHamze** [comment]
I upgraded my RVN wallet on my local PC, and then it corrupted by passphrase somehow. I have it recorded and have been using it for all previous transactions - copy paste, I dont even bother typing so no keyboard issues. 
Now I cant send anything. I've rebuilt the index, restored an older wallet data file, tried my passphrase without spaces, with question marks, tried reimporting all transactions... Looks like the RVN wallet sometimes corrupts itself. This is like coming to the bank one day and the account doesn't exist in your name anymore. The data proves you have the money (I still see all my coins in the wallet) but I have no way of sending/using them. 

I am willing to try anything anyone suggests.
